packages/teams-js/src/internal/telemetry.ts

```diff
@@ -50,6 +56,60 @@ export const enum ApiVersionNumber {
   V_3 = 'v3',
 }
 
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export class HostToAppMessageDelayTelemetry {
+  private static callbackInformation: Map<MessageUUID, CallbackInformation> = new Map();
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * Store information about a particular message.
+   * @param messageUUID The message id for the request.
+   * @param callbackInformation The information of the callback.
+   */
+  public static storeCallbackInformation(messageUUID: MessageUUID, callbackInformation: CallbackInformation): void {
+    HostToAppMessageDelayTelemetry.callbackInformation.set(messageUUID, callbackInformation);
+  }
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  public static clearMessages(): void {
+    HostToAppMessageDelayTelemetry.callbackInformation.clear();
+  }
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * Executes telemetry actions related to host to app performance metrics.
+   * @param callbackId The message id for the request.
+   * @param message The response from the host.
+   * @param logger A logger for logging any possible error.
+   */
+  public static telemetryHostToAppPerformanceMetrics(
+    callbackID: MessageUUID,
+    message: MessageResponse,
+    logger: debug.Debugger,
+  ): void {
+    const callbackInformation = HostToAppMessageDelayTelemetry.callbackInformation.get(callbackID);
+    if (callbackInformation && message.timestamp) {
```

Comment:
this addresses two way request/response API.
What about one-way API like theme changed notification from Host to App. How would that be addressed?

Reply:
Yes, you're right, I actually didn't know this feature existed until you pointed it out. Thank you!

After digging around in the codebase, I created a commit to handle the one way API cases: https://github.com/OfficeDev/microsoft-teams-library-js/pull/2499/commits/7b23b3f11f146fb9d8fca94b89a443e44f3dd734